provider/image.go

```diff
@@ -418,11 +418,22 @@ func marshalCachedImages(b resource.PropertyValue) ([]string, error) {
 
 	// if we specify a list of stages, then we only pull those
 	cacheFrom := c.ObjectValue()
+	if _, ok := cacheFrom["images"]; !ok {
```

Comment:
possibly unnecessary optimization, but could we assign `cacheFrom["images"]` to a named var to it can be reused below?

Reply:
I like it, thank you! :)